### PR TITLE
Refactor ViewModels, adapters, and feature/pickuser-viewmodel

### LINKED ISSUE #7: Allow user to invite other users to the group.

### PR DESCRIPTION
**Pull request by Puh00**
Implementation:
* ShowMembers (Group)
* AddMembers (Group)

Feature:
* PickUserViewModel+fragment for selecting users

Changes:
* Refactored ViewModels
* Refactored Adapters

Closes #7

### CODE REVIEW

**Comment by yenanw** [comment]
Pretty much done
peepoClap

**Review by OscarSanner — COMMENTED** [review]
I'll check it out asap! Just give me 7-8 hours and a gallon of coffee!

**Review by OlofSjogren — APPROVED** [review]
Will review more thoroughly in the morning, but i'm sure it's fine. This was a large one boys, perhaps keep it a bit shorter next time.

**Comment by Puh00** [comment]
The current title might be misleading, I'll change it

File: app/src/main/java/com/goayo/debtify/model/ModelEngine.java [hunk 1]
```diff
@@ -3,7 +3,6 @@
 import com.goayo.debtify.modelaccess.IGroupData;
 import com.goayo.debtify.modelaccess.IUserData;
 
-import java.util.HashSet;
```
Comment (OscarSanner) [inline app/src/main/java/com/goayo/debtify/model/ModelEngine.java:6]: smart

File: app/src/main/java/com/goayo/debtify/view/GroupFragment.java [hunk 2]
```diff
@@ -11,25 +11,29 @@
...
 import androidx.databinding.DataBindingUtil;
 import androidx.fragment.app.Fragment;
+import androidx.lifecycle.Observer;
 import androidx.lifecycle.ViewModelProviders;
 import androidx.recyclerview.widget.GridLayoutManager;
 import androidx.recyclerview.widget.RecyclerView;
 
 import com.goayo.debtify.R;
 import com.goayo.debtify.databinding.GroupFragmentBinding;
-import com.goayo.debtify.model.ModelEngine;
```
Comment (OscarSanner) [inline app/src/main/java/com/goayo/debtify/view/GroupFragment.java:20]: finally

File: app/src/main/java/com/goayo/debtify/view/ShowMembersFragment.java [hunk 3]
```diff
@@ -1,36 +1,96 @@
...
+import android.view.Menu;
 import android.view.View;
 import android.view.ViewGroup;
 
+import androidx.activity.OnBackPressedCallback;
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 import androidx.databinding.DataBindingUtil;
 import androidx.fragment.app.Fragment;
+import androidx.lifecycle.ViewModelProviders;
```
Comment (OscarSanner) [inline app/src/main/java/com/goayo/debtify/view/ShowMembersFragment.java:15]: This relates to all imports and usages of "androidx.lifecycle.ViewModelProviders". 

Is there any drawbacks to using the newer "androidx.lifecycle.ViewModelProvider" package? We've all been using this one, just raising the discussion.

**Review by OscarSanner — APPROVED** [review]

**Pull request merged**